polaris/hub/client.py

```diff
@@ -324,39 +339,54 @@ def _get_dataset(
         Returns:
             A `Dataset` instance, if it exists.
         """
-        with ProgressIndicator(
-            start_msg="Fetching artifact...",
-            success_msg="Fetched artifact.",
-            error_msg="Failed to fetch dataset.",
-        ):
-            url = (
-                f"/v1/dataset/{owner}/{name}"
-                if artifact_type == ArtifactSubtype.STANDARD.value
-                else f"/v2/competition/dataset/{owner}/{name}"
-            )
-            response = self._base_request_to_hub(url=url, method="GET")
+        url = (
+            f"/v1/dataset/{owner}/{name}"
+            if artifact_type == ArtifactSubtype.STANDARD.value
+            else f"/v2/competition/dataset/{owner}/{name}"
+        )
+        response = self._base_request_to_hub(url=url, method="GET")
 
-            # Disregard the Zarr root in the response. We'll get it from the storage token instead.
-            response.pop("zarrRootPath", None)
+        # Disregard the Zarr root in the response. We'll get it from the storage token instead.
+        response.pop("zarrRootPath", None)
 
-            # Load the dataset table and optional Zarr archive
-            with StorageSession(self, "read", Dataset.urn_for(owner, name)) as storage:
-                table = pd.read_parquet(storage.get_root())
-                zarr_root_path = str(storage.paths.extension)
+        # Load the dataset table and optional Zarr archive
+        with StorageSession(self, "read", Dataset.urn_for(owner, name)) as storage:
+            table = pd.read_parquet(storage.get_root())
+            zarr_root_path = storage.paths.extension
 
-            if artifact_type == ArtifactSubtype.COMPETITION:
-                dataset = CompetitionDataset(table=table, zarr_root_path=zarr_root_path, **response)
-                md5sum = response["maskedMd5Sum"]
-            else:
-                dataset = DatasetV1(table=table, zarr_root_path=zarr_root_path, **response)
-                md5sum = response["md5Sum"]
+            if zarr_root_path is not None:
+                # For V1 datasets, the Zarr Root is optional.
+                # It should be None if the dataset does not use pointer columns
+                zarr_root_path = str(zarr_root_path)
 
-            if dataset.should_verify_checksum(verify_checksum):
-                dataset.verify_checksum(md5sum)
-            else:
-                dataset.md5sum = md5sum
+        if artifact_type == ArtifactSubtype.COMPETITION:
```

Comment:
Not directly related to your PR, but I noticed we’re using `ArtifactSubtype.STANDARD.value` in other places, so:

```suggestion
        if artifact_type == ArtifactSubtype.COMPETITION.value:
```